### ISSUE TITLE
Bartender plist file gets overwritten

### ISSUE THREAD

**Issue by godbout** [lra/mackup#1126]
I've made a few personal and app syncs and I've found that Mackup is not working for MAS apps because the plist symlink gets overwritten by the app itself.

But now I'm having the same issue with Bartender, which is not a MAS app, so I'm little lost of why is this happening.

Anyone could confirm?

Using:
* Bartender 3.0.32
* macOS 10.13.3 (17D47)

**Comment by bensurtees** [comment]
I just had another report of this, but the user worked out the cause. So in his words:
"So the reason why Bartender is the only affected app is pretty simple: it runs before Google Drive mounted the file stream disk, so it couldn't find the preference file and set everything to default. "

That may be the cause of the issue for you.

**Comment by godbout** [comment]
Hi! Thanks. Will check. I'm synced using Dropbox. Unfortunately from what I know there's no way under macOS to decide which app starts before which other ones though.

**Comment by fullmetalsheep** [comment]
same thing happens even when using the local disk as a backup location.

**Comment by lucamaraschi** [comment]
Any solution for this?I am using `file_system` storage and it's not a loading priority issue...

**Comment by godbout** [comment]
I've moved to iCloud Drive which is mounted before Bartender, but I still have the issue too. The file gets overwritten and the symlink disappears.

**Comment by fharper** [comment]
I have the same issues with iCloud Drive.

Bartender 3.0.64
macOS 10.14.6 (18G84)

**Comment by godbout** [comment]
Today my Bartender started empty lol. All gone, even the license.

**Comment by fharper** [comment]
I switched to Dropbox, and it doesn't seems to work either. I had to configure again Bartender on my new laptop.

**Comment by godbout** [comment]
Yeah, it doesn't seem reliable with any sync service. Strangely my iMac was working fine, only my MacBook was acting weird. And I remember I've updated mackup some days ago. I've just updated now on my iMac and BOOM. Bartender is back to zero there too.

**Comment by lra** [comment]
Is Bartender actually overwriting the link at `Library/Preferences/com.surteesstudios.Bartender.plist`?
Can you `ls -l Library/Preferences/com.surteesstudios.Bartender.plist`?

**Comment by godbout** [comment]
Here is what I have:

```sh
lrwxr-xr-x    1 guill  staff     119 Jun 26 20:19 com.surteesstudios.Bartender.plist -> /Users/guill/Library/Mobile Documents/com~apple~CloudDocs/Mackup/Library/Preferences/com.surteesstudios.Bartender.plist
```

it's a symlink. No idea how Bartender handles that though.

**Comment by godbout** [comment]
lost again all settings, license, etc... time to remove bartender from the list of applications supported?

**Comment by lra** [comment]
Can you do a `ls -l Library/Preferences/com.surteesstudios.Bartender.plist` again now that you lost all settings to see if Bartender actually overrides the link?

**Comment by godbout** [comment]
It was still a symlink at the time, after I lost everything. I've removed it now from syncing through mackup.

**Comment by Amparose** [comment]
For me the bartender plist symlink gets replaced with a default Bartender plist when it opens. I am `mackup restore` with Bartender closed, confirming the plist is symlinked, and observing that Bartender replaces the symlink file when it loads.

But, I think this is a Bartender issue as even copying the plist that Mackup has copied from my other machine results in the same behaviour. Maybe it silently fails and restores default config if there are apps missing on the restore destination?

**Comment by robwilkerson** [comment]
It's been a minute since any conversation about this, but this is still a thing with Bartender 5. I have a symlink in place for the plist file, but the app just starts empty every time (the symlink isn't overwritten, but it also isn't read/understood). This feels like it might be a Bartender issue to me - maybe symlinks are being respected?

Going to check with that team, but thought I'd throw my experience in here in case others are seeing similar.
